internal/clients/kibana/connector.go

```diff
@@ -0,0 +1,1487 @@
+		return diag.Errorf("%s: %s", *resp.JSON404.Error, *resp.JSON404.Message)
+	}
+
+	if resp.JSON401 != nil {
+		return diag.Errorf("%s: %s", *resp.JSON401.Error, *resp.JSON401.Message)
+	}
+
+	if resp.StatusCode() != 200 && resp.StatusCode() != 204 {
+		return diag.Errorf("failed to delete connector: got status [%v] [%s]", resp.StatusCode(), resp.Status())
+	}
+
+	return nil
+}
+
+func ConnectorConfigWithDefaults(connectorTypeID, plan, backend, state string) (string, error) {
+	switch connectors.ConnectorTypes(connectorTypeID) {
+
+	case connectors.ConnectorTypesDotCasesWebhook:
+		return connectorConfigWithDefaultsCasesWebhook(plan)
+
+	case connectors.ConnectorTypesDotEmail:
+		return connectorConfigWithDefaultsEmail(plan)
+
+	case connectors.ConnectorTypesDotIndex:
+		return connectorConfigWithDefaultsIndex(plan)
+
+	case connectors.ConnectorTypesDotJira:
+		return connectorConfigWithDefaultsJira(plan)
+
+	case connectors.ConnectorTypesDotOpsgenie:
+		return connectorConfigWithDefaultsOpsgenie(plan)
+
+	case connectors.ConnectorTypesDotPagerduty:
+		return connectorConfigWithDefaultsPagerduty(plan)
+
+	case connectors.ConnectorTypesDotResilient:
+		return connectorConfigWithDefaultsResilient(plan)
+
+	case connectors.ConnectorTypesDotServicenow:
+		return connectorConfigWithDefaultsServicenow(plan, backend)
+
+	case connectors.ConnectorTypesDotServicenowItom:
+		return connectorConfigWithDefaultsServicenowItom(plan)
+
+	case connectors.ConnectorTypesDotServicenowSir:
+		return connectorConfigWithDefaultsServicenowSir(plan, backend)
+
+	case connectors.ConnectorTypesDotSwimlane:
+		return connectorConfigWithDefaultsSwimlane(plan)
+
+	case connectors.ConnectorTypesDotTines:
+		return connectorConfigWithDefaultsTines(plan)
+
+	case connectors.ConnectorTypesDotWebhook:
+		return connectorConfigWithDefaultsWebhook(plan)
+
+	case connectors.ConnectorTypesDotXmatters:
+		return connectorConfigWithDefaultsXmatters(plan)
+	}
+	return plan, nil
+}
+
+// User can omit optonal fields in config JSON.
+// The func adds empty optional fields to the diff.
+// Otherwise plan command shows omitted fields as the diff,
+// because backend returns all fields.
+func connectorConfigWithDefaults[T any](plan string) (string, error) {
+	var config T
+	if err := json.Unmarshal([]byte(plan), &config); err != nil {
+		return "", err
+	}
+	customJSON, err := json.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+	return string(customJSON), nil
+}
+
+func connectorConfigWithDefaultsCasesWebhook(plan string) (string, error) {
+	var custom connectors.ConfigPropertiesCasesWebhook
+	if err := json.Unmarshal([]byte(plan), &custom); err != nil {
+		return "", err
+	}
+	if custom.CreateIncidentMethod == nil {
+		custom.CreateIncidentMethod = new(connectors.ConfigPropertiesCasesWebhookCreateIncidentMethod)
+		*custom.CreateIncidentMethod = connectors.ConfigPropertiesCasesWebhookCreateIncidentMethodPost
+	}
+	if custom.HasAuth == nil {
+		custom.HasAuth = new(bool)
```

Comment:
It might be nice to add a generic Pointer func into utils so we don't need to do this dance everywhere, like:

```
func Pointer[T any](value T) *T {
    return &value
}
```